### PR TITLE
CLI reform to support new SMS API and Branding change

### PR DESCRIPTION
**Pull request by revanth0212**
## Description

This PR supports the new SMS API introduced in https://git.corp.adobe.com/commerce-gateway/schema-management-service/pull/118

This PR also introduced new commands to delete a mesh and download a mesh.

Also in the PR, I have changed the commands to the new brand `api-manager:mesh` instead `commerce-gateway:tenant`. The brand change is not official yet, we are waiting on the branding team. Mostly it is `API Manager` but if that changes later, Ill update the PR to support the new brand.

### Create Mesh

```bash
aio api-manager:mesh:create <MESH_CONFIG_FILE>
```

image

Error if something is wrong with the mesh

image

### Get Mesh

```bash
aio api-manager:mesh:get <MESH_ID>
```

image

Get also supports new features like downloading mesh to a file

```bash
aio api-manager:mesh:get <MESH_ID> <FILE_PATH>
```

image

In case of error

image

### Update Mesh

```bash
aio api-manager:mesh:update <MESH_ID> <MESH_CONFIG_FILE>
```

image

In case of error

image

### Delete Mesh

This is a new command.

```bash
aio api-manager:mesh:delete <MESH_ID>
```

image

In case of error

image

## Related Issue

https://jira.corp.adobe.com/browse/CEXT-218

## Motivation and Context

Support the new API and Brand

## How Has This Been Tested?

Pull the latest changes from https://git.corp.adobe.com/commerce-gateway/schema-management-service/pull/118 if its not already merged. Run `yarn watch` to run the SMS locally.

Update your local `config.json` to  connect to the local SMS server instead:

```json
{
	"baseUrl": "http://localhost:8080",
	"apiKey": "graphql-onboarding-io"
}
```

Currently, the latest SMS changes are deployed on Dev3, you can test using that as well.

```json
{
	"baseUrl": "https://graph-dev3.adobe.io/api-admin",
	"apiKey": "adobeio_onboarding"
}
```

Run the commands against it to test the new code

## Screenshots (if appropriate):

Provided in the description

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by brasewel — APPROVED** [review]

File: src/classes/SchemaServiceClient.js [hunk 1]
```diff
@@ -24,69 +24,191 @@ class SchemaServiceClient {
 		this.apiKey = apiKey;
 	}
 
-	async getTenant(tenantId, organizationCode) {
-		logger.info(`OrgCode - getTenant: ${organizationCode}`);
+	async getMesh(organizationCode, projectId, workspaceId, meshId) {
 		const config = {
 			method: 'get',
-			url: `${this.schemaManagementServiceUrl}/api-admin/organizations/${organizationCode}/tenants/${tenantId}?api_key=${this.apiKey}`,
+			url: `${this.schemaManagementServiceUrl}/organizations/${organizationCode}/projects/${projectId}/workspaces/${workspaceId}/meshes/${meshId}?api_key=${this.apiKey}`,
```
Comment (dthampy) [inline src/classes/SchemaServiceClient.js:30]: why did we decide to remove the api-admin here ? config.json currently looks like this:
```{
    "baseUrl": "https://commerce-beta.adobe.io",       ---> just the baseUrl without the /api-admin
    "apiKey": "commerce-graphql-onboarding"
}
```
Did we decide to change the config.json to have  baseUrl with  /api-admin appended ? If so, we will have to change [here as well](https://github.com/adobe/aio-cli-plugin-commerce-admin/pull/12/files#diff-49ba0c85360693b8e89ae0e0428284a55db749f08f2b9993d26910def646e02fL42)

**Review by dthampy — COMMENTED** [review]
Great work @revanth0212 !
Looks good to me except for that one comment I have left.

File: README.md [hunk 2]
```diff
@@ -48,6 +48,6 @@ aio config:set aio-cli-plugin-commerce-admin <path_to_json_file>
 
 ```
 aio commerce-gateway:tenant:create BODYJSONFILE
-aio commerce-gateway:tenant:update tenantid BODYJSONFILE
-aio commerce-gateway:tenant:get tenantid
+aio commerce-gateway:tenant:update meshId BODYJSONFILE
+aio commerce-gateway:tenant:get meshId
 ```
```
Comment (dthampy) [inline README.md:53]: Are we not adding the new feature you introduced to write the mesh to a file ?

**Review by dthampy — APPROVED** [review]

File: src/helpers.js [hunk 3]
```diff
@@ -65,52 +78,106 @@ async function getCommerceAdminConfig() {
...
+	if (projects.length !== 0) {
+		const selectedProject = await consoleCLI.promptForSelectProject(projects);
+
+		return selectedProject;
+	} else {
+		aioConsoleLogger.error(`No projects found for the selected organization: ${imsOrgTitle}`);
+	}
+}
+
+async function getWorkspaces(orgId, projectId, imsOrgTitle, projectTitle) {
```
Comment (amolina-magento) [inline src/helpers.js:121]: The function names are plural and suggest that more than 1 org/project/workspace will be returned. Thoughts on renaming these to `getSelectedOrganization`, `getSelectedProject` and `getSelectedWorkspace` respectively?

**Review by amolina-magento — CHANGES_REQUESTED** [review]

**Review by brasewel — CHANGES_REQUESTED** [review]
@revanth0212 I just noticed that build is currently broken which was caused by your eslint changes here
https://github.com/adobe/aio-cli-plugin-commerce-admin/commit/63dec09590334f03150392695f57cfba66a9c5a7
You can view the build here: 
https://commerce-gateway.ci.corp.adobe.com/job/aio-cli-plugin-commerce-admin/job/develop/14/console

File: src/commands/api-manager/mesh/get.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+					} catch (error) {
+						this.log('Unable to write mesh to file %s', args.file);
+
+						logger.error(error);
+					}
+				}
+
+				return mesh;
+			} else {
+				this.error(`Unable to get mesh with the ID ${mesh.meshId}`);
```
[the remark this answers is not in the document]
Comment (amolina-magento) [reply src/commands/api-manager/mesh/get.js:56]: Isn't this error visible to the user? If so, I think it's an easy change to simply omit the id from the message.

**Comment by dthampy** [comment]
I re ran the build so to make sure it picks up all the latest commits. From the build results, the failures are from tests. so the linting errors are taken care of
 https://commerce-gateway.ci.corp.adobe.com/job/aio-cli-plugin-commerce-admin/job/develop/15/

File: README.md (hunk 2 above)
Comment (revanth0212) [reply README.md:53]: Yeah, forgot to update these. Thanks guys. Gonna take care of this.

File: src/classes/SchemaServiceClient.js (hunk 1 above)
Comment (revanth0212) [reply src/classes/SchemaServiceClient.js:30]: Ideally `/api-admin` will not be part of every SMS server. It just happens to be part of our current URLs. So lets not hard code it into the code. `baseUrl` should be SMS URL and every other path that we hard code in the code like getting `mesh` etc should still work.

File: src/commands/api-manager/mesh/create.js [hunk 5]
```diff
@@ -19,27 +19,47 @@ class CreateCommand extends Command {
...
+		try {
+			const mesh = await schemaServiceClient.createMesh(imsOrgCode, projectId, workspaceId, data);
+
+			if (mesh) {
+				this.log('Successfully created mesh %s', mesh.meshId);
+				this.log(JSON.stringify(mesh, null, 2));
+
+				return mesh;
+			} else {
+				this.error(`Unable to create a mesh with the ID ${mesh.meshId}`);
```
[the remark this answers is not in the document]
Comment (revanth0212) [reply src/commands/api-manager/mesh/create.js:51]: Good point. Fixed 👍 

Ideally we will never reach this code 99% of the time but Ill leave it there for good measure.

File: src/commands/api-manager/mesh/get.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (revanth0212) [reply src/commands/api-manager/mesh/get.js:56]: Good points guys.

Made a quick change, should handle it. 99% of the time the code will not end up in that else case but just for semantic purposes I added that else part. Should be good now.

File: src/classes/logger.js [hunk 6]
```diff
@@ -2,7 +2,7 @@ const { default: pino } = require('pino');
 
 const logger = pino({
 	level: process.env.LOG_LEVEL || 'info',
-	enabled: process.env.NODE_ENV !== 'test',
+	enabled: process.env.NODE_ENV !== 'prod',
```
[the remark this answers is not in the document]
Comment (revanth0212) [reply src/classes/logger.js:5]: Enabling logger introduces a lot of noise. For instance check this

Logger Disabled
image

Logger Enabled
image

We don't want that noise while using CLI. Its definitely helpful for development. Thats why its disabled in `prod`.

[the remark this answers is not in the document]
Comment (amolina-magento) [reply src/classes/logger.js:5]: 👍

**Review by amolina-magento — APPROVED** [review]

**Comment by revanth0212** [comment]
Updates tests, nothing failing locally

File: src/commands/api-manager/mesh/__tests__/delete.test.js [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+	});
+
+	test('delete-mesh-with-meshId', async () => {
+		jest
+			.spyOn(SchemaServiceClient.prototype, 'deleteMesh')
+			.mockImplementation(meshId => mockGetMesh);
+		expect.assertions(1);
+		const meshId = 'sample_merchant';
+		const runResult = DeleteCommand.run([meshId]);
+		await expect(runResult).resolves.toEqual(mockGetMesh);
```
Comment (dthampy) [inline src/commands/api-manager/mesh/__tests__/delete.test.js:80]: If the delete is successful, are we not getting the empty result back?

**Review by dthampy — APPROVED** [review]

File: src/commands/api-manager/mesh/__tests__/delete.test.js (hunk 7 above)
Comment (revanth0212) [reply src/commands/api-manager/mesh/__tests__/delete.test.js]: Correct. Ill have to update the tests in the future PRs. Will add more tests in that PR. Fixed this for now.

Comment (dthampy) [reply src/commands/api-manager/mesh/__tests__/delete.test.js]: Thanks @revanth0212

**Pull request merged**